com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AtlasUtil.java

```diff
@@ -169,19 +165,58 @@ private interface PathTransformer {
         String transform(String path);
     }
 
+    private static String[] getReplaceTokens(String pattern) {
+        // "=" -> 0 tokens
+        // "abc=" -> 1 token:  ["abc"]
+        // "=abc" -> 2 tokens: ["", "abc"]
+        // "abc==" -> 1 token: ["abc"]
+        // "abc=def=" -> 2 tokens: ["abc", "def"]
+        String tokens[] = pattern.split("=");
```

Comment:
What happens if the string is malformed?

"hat=cat,foo=,bar"

* In the first case we get token[0] = "hat" and token[1] = "cat" - any occurrence of "hat" is replaced with "cat" 👍 
* In the second case we get token[0] = "foo" and token[1] = "" - any occurence of "foo" is replaced with "" 👍 
* In the third case we get token[0] = "" and token[1] = "" - will have no effect 👍 (it will actually early out on line 186, although I don't think it is required)

I guess there is no real problem here, but maybe we want to check that the string contains an equal character?